spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/discovery/KubernetesDiscoveryProperties.java

```diff
@@ -175,100 +177,25 @@ public String toString() {
 	}
 
 	/**
-	 * Metadata properties.
+	 * @param addLabels include labels as metadata
+	 * @param labelsPrefix prefix for the labels
+	 * @param addAnnotations include annotations as metadata
+	 * @param annotationsPrefix prefix for the annotations
+	 * @param addPorts include ports as metadata
+	 * @param portsPrefix prefix for the ports, by default it is "port."
 	 */
-	public static class Metadata {
```

Comment:
I am going to split the refactor of this class into two PRs, otherwise it might get more difficult to follow, imho. 

So, the general idea is easy : let's take this class to a `record`.